examples/common.c

```diff
@@ -391,3 +395,61 @@ int cred_acquire_cb(git_cred **out,
 	free(pubkey);
 	return error;
 }
+
+char *read_file(const char *path) {
+	FILE *file;
+	char *result;
+	size_t pos = 0;
+	size_t capacity = READ_SIZE;
+	size_t blocksRead= 0;
+	size_t totalRead = 0;
+
+	file = fopen(path, "r");
+
+	if (!file) {
+		return NULL;
+	}
+
+	result = malloc(sizeof(char) * capacity + 1);
+
+	if (result == NULL) {
+		return NULL;
+	}
+
+	while(1) {
```

Comment:
I think you could simplify that to :

```c
size_t pos = 0, blockReads = 1;
while (pos += fread(results + pos, BUFSIZE, 1, file) && !ferror(file) && !feof(file)) {
    blockReads += 1;
    result = realloc(result, sizeof(result) * ((++blocksRead)) + 1);
    …
}
```

You're also reading `READ_LEN` objects of length 1, instead of 1 object of `READ_LEN`. I'm not sure it really matters for performance, but it's still… strange 😉.

Reply:
there will be a problem with ferror if that doesn't happen at first (like read half of the file, no?)

I agreed your code is more readable but not able to replace as suggested

Reply:
My code is missing a follow-up `if (ferror(file)) /* cleanup */;` as well, so there's that. I don't think it can happen: `pos` will have advanced anyways, and the `realloc` only needs to happen if we read a full block.